examples/shader/overriding_material_shader.rs

```diff
@@ -0,0 +1,148 @@
+const FRAGMENT_SHADER: &str = r#"
+#version 450
+
+layout(location = 0) in vec2 v_Uv;
+
+layout(location = 0) out vec4 o_Target;
+
+layout(set = 1, binding = 0) uniform ColorMaterial_color {
+    vec4 Color;
+};
+
+# ifdef COLORMATERIAL_TEXTURE 
+layout(set = 1, binding = 1) uniform texture2D ColorMaterial_texture;
+layout(set = 1, binding = 2) uniform sampler ColorMaterial_texture_sampler;
+# endif
+
+void main() {
+    //Color is the color we receive from ColorMaterial's color field, we send it to the fragment shader from the vertex shader
+    vec4 color = Color;
+# ifdef COLORMATERIAL_TEXTURE
+    //Get the color of the current fragment from the texture
+    vec4 texture_pixel_color =  texture(
+        sampler2D(ColorMaterial_texture, ColorMaterial_texture_sampler),
+        v_Uv);
+    //If the color is transparent
+    if(texture_pixel_color.a == 0.0){
+        //Get the fragment's pixel position in the texture
+        vec2 pixel = v_Uv * textureSize(sampler2D(ColorMaterial_texture, ColorMaterial_texture_sampler),0);
+        //If the pixel's position fits in the equation y=x+b where b%2==0  (a straight line of pixels from every even row, creates a checkerboard pattern)
+        if(mod(int(pixel.y)-int(pixel.x),2.0)==0.0){
+            //Color the fragment slightly white
+            texture_pixel_color = vec4(0.502,0.502,0.502,1.0);
+        }
+        else{
+            //Color the fragment even whiter
+            texture_pixel_color = vec4(0.802,0.802,0.802,1.0);
+        }
+    }
+    //Mutliply the ColorMaterial's color with: 
+    //  If the alpha is zero, by our chckerboard pattern
+    //  If the alpha is non-zero, by the texture's pixel color
+    color *= texture_pixel_color;
+# endif
+    o_Target = color;
+}
+"#;
+fn setup(
+    mut commands: Commands,
+    mut render_pipelines: ResMut<Assets<PipelineDescriptor>>,
+    mut shaders: ResMut<Assets<Shader>>,
+    mut textures: ResMut<Assets<Texture>>,
+    mut materials: ResMut<Assets<ColorMaterial>>,
+) {
+    //If the sprite's original render pipeline already exists
+    if let Some(original_sprite_render_pipeline) =
+        render_pipelines.get(bevy::sprite::SPRITE_PIPELINE_HANDLE)
+    {
+        //We can clone the pipeline so we won't have to copy the code for setting it up
+        let mut pipeline_clone = original_sprite_render_pipeline.clone();
+        //In this example, we only want to override the fragment shader and so:
+        pipeline_clone.shader_stages.fragment =
+            Some(shaders.add(Shader::from_glsl(ShaderStage::Fragment, FRAGMENT_SHADER)));
+        //Adding our custom pipeline and making it untracked so it won't get removed automatically when no sprite uses it
+        render_pipelines.set_untracked(CUSTOM_SPRITE_PIPELINE_HANDLE, pipeline_clone);
+
+        //Spawning our custom sprite
+        let texture_size = 1000;
```

Comment:
Maybe the change of the shader gets clearer if we make the rendered texture smaller than the shown area (i.e. only 10x10 pixels)?

We could even put the same texture next to it using the default pipeline for comparison.

Reply:
hmm, why would it be clearer if it's smaller without another one to compare?

hmm, about the regular texture near the custom one, it's a great idea! :D hmm, I'll try to make the texture a green triangle(half the square), and the other half transparent so the pattern will be more visible like you said.
thanks for the suggestions :D